### PR TITLE
Fix incorrect return check of BN_bn2nativepad

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by kroeckx** [comment]
Could you update the documentation to say that BN_bn2lebinpad and BN_bn2nativepad also return -1 when the buffer is too small?

**Comment by PeiweiHu** [comment]
> Could you update the documentation to say that BN_bn2lebinpad and BN_bn2nativepad also return -1 when the buffer is too small?

I put them together with BN_bn2binpad. It seems they invoke the same inner interface.

**Review by kroeckx — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
This won't apply to 1.1.1. IMO it is also not worth backporting to 1.1.1 as the code fix is only for 3.0.

**Comment by kroeckx** [comment]
The documentation will apply to 1.1.1 too, I think

**Comment by t8m** [comment]
> The documentation will apply to 1.1.1 too, I think

Yes, but IMO we are not aiming to fix all documentation deficiencies in 1.1.1.

**Comment by kroeckx** [comment]
> Yes, but IMO we are not aiming to fix all documentation deficiencies in 1.1.1.

1.1.1 is our current LTS release.

**Comment by t8m** [comment]
Anyway if someone submits a 1.1.1 PR, no problem.

File: test/keymgmt_internal_test.c [hunk 1]
```diff
@@ -88,7 +88,7 @@ static int get_ulong_via_BN(const OSSL_PARAM *p, unsigned long *goal)
     int ret = 1;                 /* Ever so hopeful */
 
     if (!TEST_true(OSSL_PARAM_get_BN(p, &n))
-        || !TEST_true(BN_bn2nativepad(n, (unsigned char *)goal, sizeof(*goal))))
+        || !TEST_int_ge(BN_bn2nativepad(n, (unsigned char *)goal, sizeof(*goal)), 0))
```
Comment (paulidale) [inline test/keymgmt_internal_test.c:91]: I think this should be `TEST_int_gt()`.  A zero return on failure is being accepted as written.

**Review by paulidale — CHANGES_REQUESTED** [review]

File: test/keymgmt_internal_test.c (hunk 1 above)
Comment (PeiweiHu) [reply test/keymgmt_internal_test.c:91]: But all other invoking places use < 0 as an error and >=0 as success. 

See 
 + crypto/params.c:940
 + crypto/evp/ctrl_params_translate.c:468
 + crypto/evp/p_lib.c:2271
 + test/params_test.c:337
 + and so on

Comment (paulidale) [reply test/keymgmt_internal_test.c:91]: Yeah, you are correct.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for the contribution.

**Pull request closed without merging**